File: leetcode(04-12)/2.cpp
#include <bits/stdc++.h>
#define endl '\n'
using namespace std;

class Solution {
public:
    vector<int> processQueries(vector<int>& queries, int m) {
        int n = queries.size();
        vector<int> reff(m+1);
        for(int i=0; i<m; i++) {
            reff[i+1] = i;
        }
        vector<int> res(n);
        for(int i=0; i<n; i++) {
            res[i] = reff[queries[i]];
            for(int j=1; j<=m; j++){
                if(reff[j] < res[i])
                    reff[j] += 1;
            }    
            reff[queries[i]] = 0;
        }
        return res;
    }
};

int main()
{
    int n;
    Solution s;
    while (cin >> n)
    {
        cout << s. << endl;
    }
}

File: leetcode(04-12)/2.py
class Solution(object):
    def processQueries(self, queries, m):
        """
        :type queries: List[int]
        :type m: int
        :rtype: List[int]
        """
        ans = []
        p = list(range(1,m+1))
        for i in range( len(queries) ) :
            for j in range( len(p) ):
                if queries[i] == p[j]:
                    ans.append(j)
                    temp = p[j]
                    del p[j]
                    p.insert(0,temp)
        return ans